src/core/automount.c

```diff
@@ -1038,18 +1037,6 @@ static int automount_dispatch_io(sd_event_source *s, int fd, uint32_t events, vo
                         goto fail;
                 }
 
-                /* Before we do anything, let's see if somebody is playing games with us? */
-                if (lstat(a->where, &st) < 0) {
-                        log_unit_warning_errno(UNIT(a), errno, "Failed to stat automount point: %m");
-                        goto fail;
-                }
-
-                if (!S_ISDIR(st.st_mode) || st.st_dev == a->dev_id) {
-                        log_unit_info(UNIT(a), "Automount point already unmounted?");
-                        automount_send_ready(a, a->expire_tokens, 0);
-                        break;
-                }
-
```

Comment:
> But indeed AT_NO_AUTOMOUNT looks the right approach, I wasn't aware of this flag.

I'm taking this back.

After looking into this more carefully, this flag doesn't help since this flag has no effect if the mount point has already been mounted over. And this is the case at this point.